tools/base/libboard/board_vc/board_vc.c

```diff
@@ -0,0 +1,656 @@
+	res = read_mac_info(token, buf, SYFS_MAX_SIZE);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to mac address");
+		return res;
+	}
+
+	n = (int)buf[6];
+	printf("%-29s : %d\n", "Number of MACs", n);
+	mac.byte[0] = buf[5];
+	mac.byte[1] = buf[4];
+	mac.byte[2] = buf[3];
+	mac.byte[3] = 0;
+	for (i = 0; i < n; ++i) {
+		printf("%s %-17d : %02X:%02X:%02X:%02X:%02X:%02X\n",
+			"MAC address", i, buf[0], buf[1], buf[2],
+			mac.byte[2], mac.byte[1], mac.byte[0]);
+		mac.dword += 1;
+	}
+
+	return res;
+}
+
+// print board information
+fpga_result print_board_info(fpga_token token)
+{
+	fpga_result res                     = FPGA_OK;
+	char bmc_ver[SYFS_MAX_SIZE]         = { 0 };
+	char max10_ver[SYFS_MAX_SIZE]       = { 0 };
+	char pcb_ver[SYFS_MAX_SIZE]         = { 0 };
+
+	res = read_bmcfw_version(token, bmc_ver, SYFS_MAX_SIZE);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed bmc version");
+	}
+
+	res = read_max10fw_version(token, max10_ver, SYFS_MAX_SIZE);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to Destroy Object");
+	}
+
+	res = read_pcb_info(token, pcb_ver, SYFS_MAX_SIZE);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to Destroy Object");
+	}
+
+
+	printf("Max10 version: %s \n", max10_ver);
+	printf("BMC version: %s \n", bmc_ver);
+	printf("PCB version: %s \n", pcb_ver);
+
+	return res;
+}
+
+// print phy group information
+fpga_result print_phy_info(fpga_token token)
+{
+	fpga_result res                            = FPGA_OK;
+	fpga_phy_group_info* phy_info_array        = NULL;
+	int group_num                              = 0;
+	int fpga_mode                              = 0;
+	int i                                      = 0;
+	int j                                      = 0;
+	char mode[VER_BUF_SIZE]                    = { 0 };
+	fpga_pkvl_info pkvl_info;
+
+
+	res = read_phy_group_info(token, NULL, &group_num);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to phy group count");
+		return res;
+	}
+
+	phy_info_array = calloc(sizeof(fpga_phy_group_info), group_num);
+	if (phy_info_array == NULL) {
+		OPAE_ERR(" Failed to allocate memory");
+		return FPGA_NO_MEMORY;
+	}
+
+
+	res = read_phy_group_info(token, phy_info_array, &group_num);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to phy group array");
+		goto out_free;
+	}
+
+	res = read_pkvl_info(token, &pkvl_info, &fpga_mode);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to read pkvl info");
+		goto out_free;
+	}
+
+	//TODO
+	for (i = 0; i < group_num; i++) {
+
+		printf("//****** PHY GROUP %d ******//\n", i);
+		printf("%-29s : %s\n", "Direction",
+			phy_info_array[i].group_id == 0 ? "Line side" : "Fortville side");
+		printf("%-29s : %d Gbps\n", "Speed", phy_info_array[i].speed);
+		printf("%-29s : %d\n", "Number of PHYs", phy_info_array[i].phy_num);
+	}
+
+	//TODO
+	int mask = 0;
+	if (phy_info_array[0].speed == 10) {
+		mask = 0xff;
+
+	} else if (phy_info_array[0].speed == 25) {
+
+
+		if (phy_info_array[i].phy_num == 4) {
+			switch (fpga_mode) {
+			case 1: /* 4x25g */
+			case 3: /* 6x25g */
```

Comment:
You'll need to place FALLTHROUGH here in the comment, or else gcc on Fedora will complain.
```c
case 3: /* FALLTHROUGH 6x25g */
```

Reply:
?